docs/basics/101-125-config.rst

```diff
@@ -0,0 +1,327 @@
+
+The command consists of the base command :command:`git config`,
+a specification of the scope of the configuration with the ``--local``
+flag, a ``name`` specification consisting of section and key with the
+notation ``section.variable`` (here: ``core.editor``), and finally the value
+specification ``"nano"``.
+
+Let's see what has changed:
+
+.. runrecord:: _examples/DL-101-125-103
+   :language: console
+   :workdir: dl-101/DataLad-101
+   :emphasize-lines: 7
+
+   $ cat .git/config
+
+With this additional line in your repositories Git configuration, nano will
+be used as a default editor regardless of the configuration in your global
+or system-wide configuration. Note that the flag ``--local`` applies the
+configuration to your repository's ``.git/config`` file, whereas ``--global``
+would apply it as a user specific configuration, and ``--system`` as a
+system-wide configuration.
+If you would want to change this existing line in your ``.git/config``
+file, you would replace ``--add`` with ``--replace-all`` such as in::
+
+   git config --local --replace-all core.editor "vim"
+
+to configure :term:`vim` to be your default editor.
+
+(Note that while being a good toy example, it is not a common thing to
+configure repository-specific editors)
+
+This example demonstrated the structure of a :command:`git config`
+command. By specifying the ``name`` option with ``section.variable``
+(or ``section.subsection.variable`` if there is a subsection), and
+a value, one can configure Git, Git-annex, and DataLad.
+*Most* of these configurations will be written to a ``config`` file
+of Git, depending on the scope (local, global, system-wide)
+specified in the command.
+
+.. container:: toggle
+
+   .. container:: header
+
+       **Addition: If things go wrong**
+
+   If something goes wrong during the :command:`git config` command,
+   for example you end up having two keys of the same name because you
+   added a key instead of replacing an existing one, you can use the
+   ``--unset`` option to remove the line. Alternatively, you can also open
+   the config file in an editor and remove or change sections by hand.
+
+
+The only information you need, therefore, is the name of a section and
+variable to configure, and the value you want to specify.
+
+The configuration file inside of ``DataLad-101`` does not appear to be
+intuitively understandable enough to confidently apply changes to them,
+or identify necessary changes. And indeed, most of the sections and variables
+or values in there are irrelevant for understanding the book, your dataset,
+or DataLad, and can just be left as they are. The previous section merely served
+to de-mystify the :command:`git config` command and the configuration files.
+Nevertheless, it might be helpful to get an overview about the meaning of the
+remaining sections in that file, and the following hidden section can give
+you a glimpse of this.
+
+.. container:: toggle
```

Comment:
Note to self: This would be way to long to be typeset as a marginnote in the PDF. Maybe it needs yet another directive to distinguish between short notes and longer blocks....